File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Arrays.sort(nums);
        if(nums.length==0) return false;
         int next=nums[0];
         for(int i=1;i<nums.length;i++){
            if(next == nums[i]) return true;
            next=nums[i];
         }
        return false;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        n=len(nums)-1
        nums.sort()
        res=[]
        for i in range(n):
            l=i+1
            r=n
            while l < r:
                cur=[nums[l],nums[r],nums[i]]

                if sum(cur)==0:
                    if  cur not in res:
                            res.append(cur)
                        
                if 0 > sum(cur) : l+=1
                else: r-=1
            
        return res
                